Assets/_Scripts/Units/AI/AIEntity.cs

```diff
@@ -144,6 +145,7 @@ public override void Spawned()
 
         public override void Despawned(NetworkRunner runner, bool hasState)
         {
+            if (!NetworkSystem.Instance.IsHost) return;
```

Comment:
Ce qui est en dessous devrait être call pareil sur le client, parce que ces deux éléments (le AIManager et l'event) devraient être sync sur tous les clients et le host

Qu'est-ce qui t'amène à faire ce check?

Reply:
le message qui dit que le teacher est null ou le teacher n'est pas le teacher sur les clients. j'assume que le host fait l'action que c'est sync et quand le client le fait il essaye d'enlever le teacher qui a déjà été enlever and such.

Reply:
Il y a pas de mécanisme de synchronisation ans le AIManager en tant que tel, puisque c'est un singleton non Networked. La synchronisation se fait à partir du Spawn et du Despawned des AIs.

Je pense que ce qui se passe c'est que, dans le client, le nouveau teacher est spawned avant que l'ancien teacher (du lobby) soit despawned, ce qui donne le message de warning. Je pense que la meilleure façon de se débarasser de ce warning c'est de............. supprimer la ligne qui fait le warning 😅 (ça cause vraiment pas de bogue et elle était juste là pour m'aider à déboguer au début quand on n'avait pas de transition de scène)